server/src/test/java/org/opentosca/toscana/plugins/cloudfoundry/CloudFoundryServiceTest.java

```diff
@@ -52,40 +72,116 @@ public void setUp() {
+
+        CloudFoundryInjectionHandler injectionHandler = new CloudFoundryInjectionHandler(fileAccess, myApp);
+        Boolean isPushed = injectionHandler.deploy();
+        
+        assertTrue(isPushed);
+    }
+
+    @Test
+    public void checkReadingServiceCredentials() throws Exception {
+        assumeNotNull(envUser, envHost, envOrga, envPw, envSpace);
+        JSONObject env = null;
         try {
-            cloudFoundryConnection = new CloudFoundryConnection(envUser,
-                envPw,
-                envHost,
-                envOrga,
-                envSpace);
-
-            provider = new CloudFoundryProvider(CloudFoundryProvider.CloudFoundryProviderType.PIVOTAL);
-            provider.setOfferedService(cloudFoundryConnection.getServices());
+            env = cloudFoundryConnection.getServiceCredentials("cleardb", "my_app");
+        } catch (JSONException e) {
+            e.printStackTrace();
+            assumeTrue(false);
         } catch (Exception e) {
             assumeTrue(false);
         }
+        assertThat(env.toString(), CoreMatchers.containsString("username"));
+    }
 
-        app.setProvider(provider);
+    @Test
+    public void checkServiceCredentialInjection() throws Exception {
+        assumeNotNull(envUser, envHost, envOrga, envPw, envSpace);
+        setUpMyApp();
 
-        File sourceDir = new File(tmpdir, "sourceDir");
+        CloudFoundryInjectionHandler injectionHandler = new CloudFoundryInjectionHandler(fileAccess, myApp);
+        injectionHandler.deploy();
+
+        injectionHandler.getServiceCredentials();
+
+        Map<String, String> environmentVariables = myApp.getEnvironmentVariables();
+        String contentEnvVariables = "";
+        for (Map.Entry<String, String> entry : environmentVariables.entrySet()) {
+            contentEnvVariables = String.format("%s%s: %s\n", contentEnvVariables, entry.getKey(), entry.getValue());
+        }
+
+        assertThat(contentEnvVariables, not(CoreMatchers.containsString("database_user: TODO")));
+    }
+
+    private CloudFoundryConnection createConnection() {
+        assumeNotNull(envUser, envHost, envOrga, envPw, envSpace);
+        cloudFoundryConnection = new CloudFoundryConnection(envUser,
+            envPw,
+            envHost,
+            envOrga,
+            envSpace);
+
+        return cloudFoundryConnection;
+    }
+
+    private void setUpMyApp() throws IOException, JSONException {
+        CloudFoundryNodeVisitor visitor = new CloudFoundryNodeVisitor(myApp);
+        EffectiveModel lamp = TestEffectiveModels.getLampModel();
+        ArrayList<String> paths = new ArrayList<>();
+        String resourcesPath = "src/test/resources/";
+        File sourceDir = new File(resourcesPath + "csars/yaml/valid/lamp-input/");
         targetDir = new File(tmpdir, "targetDir");
         sourceDir.mkdir();
         targetDir.mkdir();
+        when(log.getLogger(any(Class.class))).thenReturn(LoggerFactory.getLogger("Test logger"));
```

Comment:
There is a util method that allows mocking of the Log Class:
https://github.com/StuPro-TOSCAna/TOSCAna/blob/master/server/src/test/java/org/opentosca/toscana/core/testutils/TestUtils.java